tools/contrib/modify_singlept_site_neon.py

```diff
@@ -0,0 +1,435 @@
+    surf_dir = args.surf_dir
+    surf_file = find_surffile (surf_dir, site_name) 
+
+    #--  directory structure
+    current_dir = os.getcwd()
+    parent_dir = os.path.dirname(current_dir) 
+    clone_dir = os.path.abspath(os.path.join(__file__ ,"../../.."))
+    neon_dir = os.path.join(clone_dir,"neon_surffiles")
+    print("Present Directory", current_dir) 
+
+    #--  download neon data if needed
+    neon_file = get_neon(neon_dir, site_name)
+
+    #-- Read neon data
+    df = pd.read_csv (neon_file)
+
+    # -- Read surface dataset files
+    f1 = xr.open_dataset(surf_file)
+
+    # -- Find surface dataset soil depth information
+    soil_bot, soil_top = find_soil_structure (surf_file)
+
+    # -- Find surface dataset soil levels
+    # TODO: how? NS uses metadata on file to find 
+    # soil strucure
+    # better suggestion by WW to write dzsoi to neon surface dataset
+
+    print (soil_top)
+    print ("Sum of soil top depths    :", sum(soil_top))
+    print (soil_bot)
+    print ("Sum of soil bottom depths :",sum(soil_bot))
+
+    soil_top = np.cumsum(soil_top)
+    soil_bot = np.cumsum(soil_bot)
+    soil_mid = 0.5*(soil_bot - soil_top)+soil_top
+    print ("Cumulative sum of soil bottom depths :", sum(soil_bot))
+    # TODO Will: if I sum them up , are they 3.5? (m)
+
+    obs_top = df['biogeoTopDepth']/100
+    obs_bot = df['biogeoBottomDepth']/100
+
+    # -- Mapping surface dataset and neon soil levels
+    bins = df['biogeoTopDepth']/100
+    bin_index = np.digitize(soil_mid, bins)-1
+
+
+    '''
+    print ("================================")
+    print ("  Neon data soil structure:     ")
+    print ("================================")
+
+    print ("------------","ground","------------")
+    for i in range(len(obs_bot)):
+        print ("layer",i)
+        print ("-------------",
+                "{0:.2f}".format(obs_bot[i]),
+                "-------------")
+
+    print ("================================")
+    print ("Surface data soil structure:    ")
+    print ("================================")
+
+    print ("------------","ground","------------")
+    for b in range(len(bin_index)):
+        print ("layer",b)
+        print ("-------------",
+                "{0:.2f}".format(soil_bot[b]),
+                "-------------")
+    '''
+    #-- update fields with neon
+    f2= f1
+    soil_levels = f2['PCT_CLAY'].size
+    for soil_lev in range(soil_levels):
+        #print (soil_lev)
+        f2['PCT_CLAY'][soil_lev] = df['clayTotal'][bin_index[soil_lev]]
+        f2['PCT_SAND'][soil_lev] = df['sandTotal'][bin_index[soil_lev]]
+        bulk_den = df['bulkDensExclCoarseFrag'][bin_index[soil_lev]]
+        carbon_tot = df['carbonTot'][bin_index[soil_lev]]
+        #print ("carbon_tot:", carbon_tot)
+        layer_depth = df['biogeoBottomDepth'][bin_index[soil_lev]] - df['biogeoTopDepth'][bin_index[soil_lev]]
+        f2['ORGANIC'][soil_lev] = carbon_tot *  bulk_den * 0.1 / layer_depth * 100 / 0.58 
```

Comment:
maybe plot up old and new organic values for a few sites to check that values seem to be of similar magnitude?

Reply:
Hi @negin513 sorry this has been so confusing.  Here's what I think we need do for ORGANIC on the surface dataset.  We're taking these data from NEON: 
- `carbonTot` (gC/kg soil) and 
- `bulkDensExclCoarseFrag` (g soil / cm3 soil) and need to calculate
- `ORGANIC` (kg OM/m3 soil, assuming 0.58gC/gOM)

To do this I think we just need:
- ORGANIC = carbonTot * bulkDensity / 0.58
- I think all the units converting between g-kg and cm3-m3 end up dropping out
- I also don't think we need depth, since units for ORGANIC are volumetric, not on an areas basis.

After doing this calculation with the raw NEON data, then you can select the CLM soil layer that corresponds to the NEON horizon, as you did for SAND and CLAY.
 - OR - 
We can do a depth weighted average for values between NEON layers, unlike for texture.

Reply:
sorry for confusion on this and I can provide more details, but it seems like the equation here should be: 

ORGANIC = carbonTot * bulk_den / 0.58 (units would be kgOM/m3 soil)

Then we can select CLM layers that correspond to the matching NEON observations?